Editor/Windows/FolderPanel.cs

```diff
@@ -171,7 +206,7 @@ public void EnsureFolderExists()
             for (int i = 0; i < listProperty.arraySize; i++)
             {
                 PaletteFolder folder = listProperty.GetArrayElementAtIndex(i).GetValue<PaletteFolder>();
-                if (folder.Name == desiredName)
+                if (folder != null && string.Equals(folder.Name, desiredName, StringComparison.Ordinal))
```

Comment:
Same question as before: what is the null check for? I don't think it should be possible for this to be null ever.

Were you getting null reference exceptions here?

Reply:
Same answer than before 😆 

When creating a new folder with the context menu on the Folder Panel, while you still have the focus to rename the folder, this was throwing Null References for me.

Reply:
Strange edge case, but if it works now then I'm cool with it 😄